pkg/app/piped/trigger/determiner.go

```diff
@@ -50,13 +50,31 @@ func NewDeterminer(repo git.Repo, targetCommit string, cg LastTriggeredCommitGet
 }
 
 // ShouldTrigger decides whether a given application should be triggered or not.
-func (d *Determiner) ShouldTrigger(ctx context.Context, app *model.Application) (bool, error) {
+// Flag `ignoreUserConfig` set to `true` will force check changes and use it to determine
+// the application deployment should be triggered or not, regardless of the user's configuration.
+func (d *Determiner) ShouldTrigger(ctx context.Context, app *model.Application, ignoreUserConfig bool) (bool, error) {
```

Comment:
How about moving `ignoreUserConfig` to Determiner while initializing it? In that way, we can keep this function simple as it is.

Reply:
Sure, lets me adopt that 🙆‍♀️

Reply:
addressed by https://github.com/pipe-cd/pipe/pull/2714/commits/8dffc8dde1ce421ecde90d24f915b6589f3b63db